### PR TITLE
fix(dedicated): server names getting back to old name

### PR DESCRIPTION
**Pull request by vks-ovh**
ref: DTRSD-88132

  Adddition of If-modified-since header to handle random update of server name.

Signed-off-by: vikash singh <vikash.singh@corp.ovh.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #DTRSD-88132
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- ~~[ ] Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- ~~[ ] Breaking change is mentioned in relevant commits~~

## Description

Addition of 'if-modified-since' header to handle the random update of server name.

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js [hunk 1]
```diff
@@ -2213,4 +2213,17 @@ export default class ServerF {
       .get(`/dedicated/server/${serviceName}/bringYourOwnImage`)
       .then(({ data }) => data);
   }
+
+  getServices(modifiedSince) {
+    return this.OvhHttp.get('/service', {
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js:2218]: If duable prefer to use `$http`.

File: packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js [hunk 2]
```diff
@@ -2213,4 +2213,17 @@ export default class ServerF {
...
+      rootPath: '2api',
+      params: {
+        external: false,
+        type: '/dedicated/server',
+      },
+      headers: {
+        'If-Modified-Since': modifiedSince,
```
Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js:2225]: Why not directly send a 'no-cache' header and check if it is forwarded

Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js:2225]: Why not directly send a 'no-cache' header and check if it is forwarded

**Review by marie-j — CHANGES_REQUESTED** [review]

File: packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js (hunk 1 above)
Comment (vks-ovh) [reply packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js:2218]: Its not, I tried.

**Review by ghyenne — DISMISSED** [review]

**Review by sachinrameshn — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js (hunk 1 above)
Comment (antleblanc) [reply packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js:2218]: This code snippet isn't working?

```js
return this.$http.get('URL', {
  params: {
    external: false,
    type: '/dedicated/server',
   },
   serviceType: 'aapi',
   headers: {
     pragma: 'no-cache',
   },
  });
```

Comment (vks-ovh) [reply packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js:2218]: @antleblanc yeah, i tried the same it didn't work. But let me give it one more shot.

Comment (vks-ovh) [reply packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js]: Taken care.

**Review by ghyenne — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Review by marie-j — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by antleblanc — APPROVED** [review]

**Pull request merged**